apps/backend/src/applications/applications.service.ts

```diff
@@ -67,61 +75,89 @@ export class ApplicationsService {
         key,
         appStatus,
       );
+      if (appStatus === ApplicationStatus.APPROVED) {
+        const user = await this.userService.getUser(application.userId);
+        const site = await this.siteService.getSite(application.siteId);
+        const name = user.firstName;
+        const email = user.email;
+
+        const siteName = site.siteName;
+        const timeFrame = '30 days';
+        const emailData = {
+          firstName: name,
+          userEmail: email,
+          siteName: siteName,
+          timeFrame: timeFrame,
+        };
+        const lambdaResult = await this.lambdaService.invokeLambda(
+          'giSendApplicationApproved',
+          emailData,
+        );
+        console.log('Lambda result: ', lambdaResult);
+      }
       return this.mapDynamoDBItemToApplication(updatedApplication);
     } catch (e) {
       throw new Error('Unable to update application status: ' + e);
     }
   }
- 
+
   public async postApplication(applicationData: NewApplicationInput) {
-    const newId = await this.dynamoDbService.getHighestAppId(this.tableName) + 1;
-    const applicationModel = this.PostInputToApplicationModel(applicationData, newId.toString());
-    console.log("Received application data:", applicationData);
+    const newId =
+      (await this.dynamoDbService.getHighestAppId(this.tableName)) + 1;
+    const applicationModel = this.PostInputToApplicationModel(
+      applicationData,
+      newId.toString(),
+    );
+    console.log('Received application data:', applicationData);
     try {
-        const result = await this.dynamoDbService.postItem(this.tableName, applicationModel);
-
-        if (result.$metadata.httpStatusCode !== 200) {
-            throw new Error('Error posting application');
-        }
-        const user = await this.userService.getUser(applicationData.userId);
-        const site = await this.siteService.getSite(applicationData.siteId);
-        const name = user.firstName;
-        const email = user.email;
- 
-        const siteName = site.siteName;
-        const timeFrame = "30 days"
-        const emailData = {"firstName":name, "userEmail":email, "siteName":siteName, "timeFrame":timeFrame};
-
-
-
-        const lambdaResult = await this.lambdaService.invokeLambda('giSendApplicationConfirmation', emailData );
-        console.log("Lambda result: ", lambdaResult);
+      const result = await this.dynamoDbService.postItem(
+        this.tableName,
+        applicationModel,
+      );
 
+      if (result.$metadata.httpStatusCode !== 200) {
+        throw new Error('Error posting application');
+      }
+      const user = await this.userService.getUser(applicationData.userId);
+      const site = await this.siteService.getSite(applicationData.siteId);
```

Comment:
```suggestion
      const site = await this.siteService.getSite(applicationData.siteId.N);
```
Don't forget the .N to extract the number!

Reply:
Im getting this ts error when I add .N,
![image](https://github.com/user-attachments/assets/fbf8f17c-8755-4351-b1cf-d157b3fe3730)

Reply:
I added .N to the userid and siteid from the updateapplication function and that seemed to work because thats the one retrieving from dynamo